src/cargo/util/semver_eval_ext.rs

```diff
@@ -0,0 +1,595 @@
+//! This module provides the [`matches_prerelease`] to extends the semantic for how to match a pre-release version.
+//! In particular, this allows non pre-release version to update to Semver-compatible pre-release version.
```

Comment:
nit: when writing a doc comment, consider what the one-line explaination that `rustdoc` will show when browsing [`util`](https://doc.rust-lang.org/nightly/nightly-rustc/cargo/util/index.html).

So maybe something like

```suggestion
//! Extend `semver::VersionReq` with  [`matches_prerelease`] which doesn't preclude pre-releases by default
```

Reply:
I've changed it this way